File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/crypto/mocks/MockedKeyGenerator.java
package at.tugraz.tc.cyfile.crypto.mocks;

import java.security.Provider;

import javax.crypto.KeyGenerator;
import javax.crypto.KeyGeneratorSpi;

public class MockedKeyGenerator extends KeyGenerator {
    /**
     * Creates a KeyGenerator object.
     *
     * @param keyGenSpi the delegate
     * @param provider  the provider
     * @param algorithm the algorithm
     */
    public MockedKeyGenerator(KeyGeneratorSpi keyGenSpi, Provider provider, String algorithm) {
        super(keyGenSpi, provider, algorithm);
    }


}

File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/crypto/mocks/MockedKeyGenerator.kt
package at.tugraz.tc.cyfile.crypto.mocks

import java.security.Provider

import javax.crypto.KeyGenerator
import javax.crypto.KeyGeneratorSpi

class MockedKeyGenerator
/**
 * Creates a KeyGenerator object.
 *
 * @param keyGenSpi the delegate
 * @param provider  the provider
 * @param algorithm the algorithm
 */
(keyGenSpi: KeyGeneratorSpi, provider: Provider, algorithm: String) : KeyGenerator(keyGenSpi, provider, algorithm)
